File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr, int m) {
        List<Integer> ans = new ArrayList<>();
        for (int a : arr) {
            if (a % m == 0) ans.add(a);
        }
        if (ans.size() > 0) {
            Collections.sort(ans);
            return ans.stream().mapToInt(Integer::intValue).toArray();
        } else {
            return new int[] {-1};
        }
    }
}

File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.kt
class Solution {
    fun solution(arr: IntArray, d: Int): IntArray {
        val ans = mutableListOf<Int>()
        for (e in arr) {
            if (e % d == 0) ans.add(e)
        }
        if (ans.size > 0) {
            return ans.sorted().toIntArray()
        } else {
            return intArrayOf(-1)
        }
    }
}